### PR TITLE
feat(es/ast): Support import phase

### PR DESCRIPTION
**Pull request by magic-akari**
**Description:**

- https://github.com/tc39/proposal-source-phase-imports

### CODE REVIEW

**Comment by magic-akari** [comment]
How to fix node bindings?

File: crates/swc_ecma_parser/src/parser/stmt/module_item.rs [hunk 1]
```diff
@@ -81,30 +72,56 @@ impl<I: Tokens> Parser<I> {
...
+
+            // OK: import <phase> x from "foo";
+            // OK: import <phase> from from "foo";
+            // NO: import <phase> from "foo";
+            // NO: import <phase> from 'foo';
+            // OK: import <phase> { x } from "foo";
+            // OK: import <phase> * as T from "foo";
+            // NO: import <phase> "foo";
+            let is_phase = is_one_of!(self, BindingIdent, '{', '*')
+                && (!is!(self, "from") || peeked_is!(self, "from"));
```
Comment (magic-akari) [inline crates/swc_ecma_parser/src/parser/stmt/module_item.rs:111]: Is it possible to peek more than one token?
https://github.com/oxc-project/oxc/blob/891d353adeefbd54e56c2ff3e89a49697da0bc5b/crates/oxc_parser/src/js/module.rs#L457

Comment (kdy1) [reply crates/swc_ecma_parser/src/parser/stmt/module_item.rs:111]: I'll add a method to do so. Also, we need to roll-up AST-breaking changes, so I'll work on #8026 and the setter prop AST thing.

**Comment by kdy1** [comment]
For node bindings, you can do https://github.com/swc-project/swc/blob/0c05f6e76e6701902ccfde405cfa16ccca8cfa48/.github/workflows/CI.yml#L170-L175

**Comment by magic-akari** [comment]
Plugin test failed in node bindings test.
It's a breaking change since the AST changed.

IMO it's expected result, right?

**Comment by kdy1** [comment]
Yes, it's expected

**Review by kdy1 — APPROVED** [review]
Thank you! I'll merge this when other AST-breaking changes are ready.

---


swc-bump:
 - swc_common
 - swc_ecma_ast --breaking

**Review by kdy1 — APPROVED** [review]
Can you rebase it with the `dev` branch?

---

swc-bump:
 - dbg-swc

**Comment by magic-akari** [comment]
> Can you rebase it with the `dev` branch?

done.

**Pull request merged**